clientd3d/web.c

```diff
@@ -19,9 +19,34 @@ using namespace std;
 /*
  * WebLaunchBrowser:  Attempt to run browser on given URL.
  */
-void WebLaunchBrowser(char *url)
+void WebLaunchBrowser(const char *url)
 {
-  ShellExecute(NULL, "open", url, NULL, NULL, SW_SHOWNORMAL);
+    if (url == nullptr || strlen(url) == 0) {
+        debug(("URL string empty\n"));
+        return;
+    }
+
+    std::string urlStr(url);
+
+    if (ContainsSpaces(urlStr)) {
+        debug(("URL contains spaces\n"));
+        return;
+    }
+
+    if (urlStr.find("https://") != 0 && urlStr.find("https://") != 0) {
```

Comment:
Doing the same check twice?  Is one of these supposed to be http?

Reply:
Yes, mistake.  Changed to 2 separate checks (one should be HTTP, the other should be HTTPS).